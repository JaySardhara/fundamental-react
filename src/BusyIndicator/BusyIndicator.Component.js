import { BusyIndicator } from '../';
import path from 'path';
import React from 'react';
import { ComponentPage, Example } from '../_playground';

export const BusyIndicatorComponent = () => {
    return (
        <ComponentPage
            description={'A **Busy Indicator** informs the user of an ongoing operation.'}
            sourceModulePath={path.join(__dirname, './BusyIndicator')}
            title='Busy Indicator'>
            <Example centered title='Busy Indicator'>
                <BusyIndicator />
            </Example>

            <Example
                centered
                description={'There are 3 sizes for Busy Indicator: s, m & l'}
                title='Busy Indicator Sizes'>
                <BusyIndicator size='s' />
                <BusyIndicator />
                <BusyIndicator size='l' />
            </Example>
        </ComponentPage>
    );
};
