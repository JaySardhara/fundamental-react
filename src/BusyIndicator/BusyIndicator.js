import { BUSY_INDICATOR_SIZES } from '../utils/constants';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import React, { useEffect } from 'react';

const BusyIndicator = React.forwardRef(({
    children,
    className,
    size,
    disableStyles,
    ...props
}, ref) => {

    useEffect(() => {
        if (!disableStyles) {
            require('fundamental-styles/dist/icon.css');
            require('fundamental-styles/dist/busy-indicator.css');
        }
    }, []);

    const busyIndicatorClasses = classnames(
        {
            [`fd-busy-indicator--${size}`]: !!size
        },
        className
    );

    return (
        <div
            {...props}
            className={busyIndicatorClasses}
            ref={ref} >
            <div className='fd-busy-indicator--circle-0' />
            <div className='fd-busy-indicator--circle-1' />
            <div className='fd-busy-indicator--circle-2' />
        </div>
    );
});

BusyIndicator.displayName = 'BusyIndicator';

BusyIndicator.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    disableStyles: PropTypes.bool,
    size: PropTypes.oneOf(BUSY_INDICATOR_SIZES)
};

BusyIndicator.defaultProps = {
    size: 'm'
};

export default BusyIndicator;
