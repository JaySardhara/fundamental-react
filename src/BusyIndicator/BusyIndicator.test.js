import BusyIndicator from './BusyIndicator';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<BusyIndicator />', () => {
    const defaultBusyIndicator = <BusyIndicator />;
    const smallBusyIndicator = <BusyIndicator size='s' />;
    const largeBusyIndicator = <BusyIndicator size='l' />;

    test('create busy indicators', () => {
        // default busy indicator
        let component = renderer.create(defaultBusyIndicator);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // small busy indicator
        component = renderer.create(smallBusyIndicator);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // large busy indicator
        component = renderer.create(largeBusyIndicator);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the Busy Indicator component', () => {
            const element = mount(<BusyIndicator data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
    test('forwards the ref', () => {
        let ref;
        class Test extends React.Component {
            constructor(props) {
                super(props);
                ref = React.createRef();
            }
            render = () => <BusyIndicator ref={ref} />;
        }
        mount(<Test />);
        expect(ref.current.tagName).toEqual('DIV');
        expect(ref.current.className).toEqual('fd-busy-indicator--m');
    });
});
