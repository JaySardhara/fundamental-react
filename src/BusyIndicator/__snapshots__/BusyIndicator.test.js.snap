// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusyIndicator /> create busy indicators 1`] = `
<div
  className="fd-busy-indicator--m"
>
  <div
    className="fd-busy-indicator--circle-0"
  />
  <div
    className="fd-busy-indicator--circle-1"
  />
  <div
    className="fd-busy-indicator--circle-2"
  />
</div>
`;

exports[`<BusyIndicator /> create busy indicators 2`] = `
<div
  className="fd-busy-indicator--s"
>
  <div
    className="fd-busy-indicator--circle-0"
  />
  <div
    className="fd-busy-indicator--circle-1"
  />
  <div
    className="fd-busy-indicator--circle-2"
  />
</div>
`;

exports[`<BusyIndicator /> create busy indicators 3`] = `
<div
  className="fd-busy-indicator--l"
>
  <div
    className="fd-busy-indicator--circle-0"
  />
  <div
    className="fd-busy-indicator--circle-1"
  />
  <div
    className="fd-busy-indicator--circle-2"
  />
</div>
`;

exports[`<BusyIndicator /> create busy indicators 4`] = `null`;

exports[`<BusyIndicator /> hide busy indicator 1`] = `null`;
